crates/uv-git/src/git.rs

```diff
@@ -74,11 +74,7 @@ impl GitReference {
         if rev.starts_with("refs/") {
             Self::NamedRef(rev)
         } else if looks_like_commit_hash(&rev) {
-            if rev.len() == 40 {
-                Self::FullCommit(rev)
-            } else {
-                Self::BranchOrTagOrCommit(rev)
-            }
+            Self::BranchOrTagOrCommit(rev)
```

Comment:
Is this change separate from the change in this pull request?

Reply:
No, this is necessary. `Self::FullCommit` must be a commit. It's not enough to store a thing that looks like a commit.

Reply:
I thought it was guaranteed that a 40 character revision is a commit? Or is the problem that's GitHub only?

Reply:
It's specific to GitHub. Git itself allows it. We could continue to make this assumption if you're comfortable with it? (We were already fetching this commit from GitHub before so removing this isn't a performance regression; it's just "the same" as before.)

Reply:
I'm comfortable with assuming it, or not. I was mostly just confused that it changed here.